docs/tutorial/avax-nft.md

```diff
@@ -0,0 +1,399 @@
+        uint256 newItemId = _tokenIds.current();
+        _mint(player, newItemId);
+        _setTokenURI(newItemId, tokenURI);
+
+        return newItemId;
+    }
+}
+
+```
+
+You can come up with your own contract's name, token's name, and token's ticker. I'm naming mine `Filet` here.
+
+Now compile the contract `Filet.sol` using this hardhat command:
+
+```shell
+npx hardhat compile
+> Compiling 1 file with 0.8.0
+> Compilation finished successfully
+```
+
+Once the contract is compiled, create another directory named `script` and add `deploy.js` with the following code:
+
+```js
+import {
+  Contract,
+  ContractFactory
+} from "ethers"
+import { ethers } from "hardhat"
+
+const deploy = async (contractName) => {
+  const Contract = await ethers.getContractFactory(contractName)
+  const contract = await Contract.deploy()
+
+  await contract.deployed()
+  console.log(`${contractName} deployed to: ${contract.address}`)
+}
+
+const main = async () => {
+  await deploy("Filet")
+}
+
+main()
+.then(() => process.exit(0))
+.catch(error => {
+  console.error(error)
+  process.exit(1)
+})
+```
+
+This script uses [Ethers](https://docs.ethers.io/v5/) library to deploy the contract to the local Avalanche node(s). Now, deploy the contract with:
+
+```shell
+npx hardhat run scripts/deploy.js --network local
+> Filet deployed to: 0x5FbDB2315678afecb367f032d93F642f64180aa3
+```
+
+We now have a token contrat deployed at `0x5FbDB2315678afecb367f032d93F642f64180aa3`.
+
+Note that up until now, the steps are the same for both Ethereum and Avalanche.
+
+Now let's spin up Hardhat's [developer console](https://hardhat.org/guides/hardhat-console.html) to start interacting with our `Filet` contract:
+
+```shell
+npx hardhat console --network local
+> Welcome to Node.js v14.18.1.
+> Type ".help" for more information.
+> >
+```
+
+Now, type the following into the prompt to instantialize the contract object:
+
+```js
+>> const Filet = await ethers.getContractFactory("Filet")
+>> const filet = await Filet.attach("0x5FbDB2315678afecb367f032d93F642f64180aa3")
+```
+
+Next, we inspect the accounts and the balances:
+
+```js
+>> const accounts = await ethers.provider.listAccounts()
+>> accounts
+> [
+>   '0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC',
+>   '0x9632a79656af553F58738B0FB750320158495942',
+>   '0x55ee05dF718f1a5C1441e76190EB1a19eE2C9430',
+>   // ...
+> ]
+```
+
+The array, unsurprisingly, should contain all the addresses listed with the previous `npx hardhat accounts` command. Select one of the addresses to inspect the balance (In this case, we are choosing the second account listed)
+
+```js
+>> const balance = (await filet.balanceOf(accounts[1])).toString()
+> 0
+```
+
+Obviously `0x9632a79656af553F58738B0FB750320158495942` does not own any FIT token yet. Let's mint some and send to the address with:
+
+```js
+>> const tokenId = await filet.callStatic.mintTo(accounts[1], "ipfs://bafkreigaymo3qz73w4nit2matfs7dugczda5wuwzq4g3o2chz4f6nugtaq/metadata.json")
```

Comment:
I will recommend moving `Uploading NFT metadata` section up so it is easy to understand the workflow of minting an NFT.  Otherwise, I would wonder where is this IPFS url from